### PR TITLE
Adds style closet layout styles and adjustments

### PR DESCRIPTION
**Pull request by agirlnamedsophia**
Summary
* This begins the shift to fully integrate style closet. I started structuring the stylesheets according to components and pages.
* There are a few non-SUIT compliant style choices so as not to break the current view in a few pages that haven't been redesigned, and that were simply style-closeted over. I'd like feedback on how to proceed with those, but am also happy to leave them in until the next final style-transformation PR at which point they could be yanked.
* There are a few lingering bootstrap references that relate to flashes and form errors that will remain until the _next_ pr, at which point bootstrap will be removed and those references will be deleted. The reason is primarily due to the icons bootstrap gives us, and I didn't want to overwhelm this PR with a new font in addition to the layout changes. 

* In doing this work I had to adjust a few siteprism pages. I imagine when our input helpers are included and bootstrap is deprecated a lot of these spec pages can be reworked to be more flexible and not so class-name-dependent. But that's for another day.

Here's a gif to show how it looks with these layout styles added. 

/domain @Betterment/test_track_core 
testtrack

### CODE REVIEW

File: app/views/admin/bulk_assignments/new.html.erb [hunk 1]
```diff
@@ -1,45 +1,41 @@
...
-    <%= f.input :identifiers_listing, as: :text, label: 'Ids To Assign', wrapper_html: { class: "col-sm-8" }, :input_html => {:rows => 10} %>
-  </div>
-  <% if @bulk_assignment_creation.new_identifier_creation_ratio_above_warning_threshold? %>
-    <div class="row">
-      <div class="form-group has-error col-sm-6" id="identifier_creation_warning">
-        <div class="help-block has-error">
+<div class="InfoCard">
+  <%= simple_form_for(@bulk_assignment_creation, url: admin_split_bulk_assignments_path) do |f| %>
+    <div class="InfoCard-row">
+      <%= f.input :identifiers_listing, as: :text, label: 'Ids To Assign', :input_html => {:rows => 10} %>
```
Comment (dschaub) [inline app/views/admin/bulk_assignments/new.html.erb]: to be annoying: "IDs"

File: app/views/admin/decisions/new.html.erb [hunk 2]
```diff
@@ -1,39 +1,20 @@
...
-        %>
-      </div>
+    <p class="title">Deciding this split in favor of a variant will have two immediate consequences:</p>
+    <ol class="sc-m-l--s sc-p-l--xs">
+      <li>This split will be re-weighted so that all <span class="u-emphasis">future</span> assignments will be to the variant specified.</li>
+      <li>All visitors currently assigned to any other variant in this split will be <span class="u-emphasis">re-assigned</span> to the variant specified.</li>
+    </ol>
+
+    <div class="InfoCard-row sc-m-t--m">
+      <strong class="">Current Population: <%= @split.assignments.count %></strong>
```
Comment (dschaub) [inline app/views/admin/decisions/new.html.erb]: empty class?

File: app/views/admin/sessions/new.html.erb [hunk 3]
```diff
@@ -1,5 +1,4 @@
-<% content_for :body_class do %>sessions<% end %>
-<section class="row sessions">
-  <h1>Test Track</h1>
-  <%= link_to value='Log in',  admin_omniauth_authorize_path(:saml), class: 'login saml' %>
-</section>
+<div class="LoginForm">
+  <h1 class="LoginForm-title sc-m-t--xl sc-m-b--m">Test Track</h1>
+  <%= link_to value='Log in',  admin_omniauth_authorize_path(:saml), class: 'button LoginCard-cta saml' %>
```
Comment (dschaub) [inline app/views/admin/sessions/new.html.erb]: do we need the `value=` bit?

**Comment by dschaub** [comment]
<< domain tafn, I'll leave this mostly up to the scss squad

**Comment by agirlnamedsophia** [comment]
bump

**Comment by dschaub** [comment]
domain lgtm

File: app/assets/stylesheets/components/_header.scss [hunk 4]
```diff
@@ -0,0 +1,17 @@
+.sc-TakeoverHeader {
+
+  &-content {
+    @include display(flex);
+    @include justify-content(space-between);
+  }
+
+  &--multipleCta {
+    @include display(flex);
```
Comment (chrislopresto) [inline app/assets/stylesheets/components/_header.scss]: We don't need `display: flex;` here. Instead we should remove the divs wrapping the links.

File: app/assets/stylesheets/components/_header.scss [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+
+    a {
```
Comment (chrislopresto) [inline app/assets/stylesheets/components/_header.scss]: These rules are not specific to the TakeoverHeader component. If we need special link rules, let's use a class selector in some app-specific link component / utility.

Taken together, that means we should be able to ditch this `multipleCta` component variant altogethe.

File: app/views/shared/_header.html.erb [hunk 6]
```diff
@@ -0,0 +1,15 @@
+<div class="sc-TakeoverHeader-content">
+  <figure class="sc-TakeoverHeader-logoContainer sc-p--xs">
+    <%= link_to(image_tag('ttlogo.png', size: "47x47"), admin_root_url) %>
+  </figure>
+  <% if signed_in? %>
+    <div class="sc-TakeoverHeader--multipleCta">
```
Comment (chrislopresto) [inline app/views/shared/_header.html.erb]: In my comments above, I didn't realize this modifier selector wasn't on the component root element. Sticking with the component usage guidelines, we should end up with markup like this:

```erb
<div class="sc-TakeoverHeader-cta">
  <%= link_to 'Splits', admin_root_path, class: 'sc-p--s' %>
  <%= link_to 'Log out', destroy_admin_session_path, method: :delete. class: 'sc-p--s' %>
</div>
```

File: vendor/gems/style-closet/app/assets/javascripts/style_closet.js [hunk 7]
```diff
@@ -1 +1 @@
-//= require ./style_closet/
+//= require_tree .
```
Comment (chrislopresto) [inline vendor/gems/style-closet/app/assets/javascripts/style_closet.js]: I don't see any other JavaScript changes. Can we ditch this change?

File: spec/support/pages/admin/split_show_page.rb [hunk 8]
```diff
@@ -3,9 +3,10 @@ class AdminSplitShowPage < SitePrism::Page
 
   element :population_count, "tr.population-row span.population"
   element :update_assignments, "tr.population-row a"
+  element :change_weights, "a.change-weights-link"
+  element :decide_split, "a.split-decision"
```
Comment (chrislopresto) [inline spec/support/pages/admin/split_show_page.rb]: Should we just use class selectors here? Do the tests need to care whether the split decision CTA is a button or a link?

File: spec/support/pages/admin/split_index_page.rb [hunk 9]
```diff
@@ -1,6 +1,6 @@
 class AdminSplitIndexPage < SitePrism::Page
   set_url '/admin'
 
-  element :splits_table, 'body.splits.index table'
-  element :log_out, '.performer-nav a[href="/admins/sign_out"]'
+  element :splits_table, 'body.AdminSplits--index table'
+  element :log_out, 'header a[href="/admins/sign_out"]'
```
Comment (chrislopresto) [inline spec/support/pages/admin/split_index_page.rb]: Can we just remove `.performer-nav|header` from the selector? No need to be this specific, right?

File: app/helpers/application_layout_helper.rb [hunk 10]
```diff
@@ -0,0 +1,17 @@
...
+
+  def body_css_classes
+    [controller_action_css_class, controller_css_class].join(' ')
+  end
+
+  def controller_css_class
+    controller_path.tr('/', '_').camelize
+  end
+
+  def controller_action_css_class
```
Comment (chrislopresto) [inline app/helpers/application_layout_helper.rb]: Can we copy/paste the specs for these helpers, too? 😄

File: app/assets/stylesheets/components/_tables.scss [hunk 11]
```diff
@@ -0,0 +1,9 @@
+.BasicTable {
+  @include standard-table(100%, 40%, 25%);
+  max-width: 1200px;
```
Comment (chrislopresto) [inline app/assets/stylesheets/components/_tables.scss]: Do we need this `max-width`? If we do, can/should we express it as `max-width: $app-width;`?

File: app/assets/stylesheets/components/_header.scss [hunk 12]
```diff
@@ -0,0 +1,17 @@
+.sc-TakeoverHeader {
+
+  &-content {
+    @include display(flex);
```
Comment (chrislopresto) [inline app/assets/stylesheets/components/_header.scss]: `.sc-TakeoverHeader-content` already has `display: flex;` I don't _think_ we need `justify-content: space-between;` here, either. The component should do this by default.

File: app/assets/stylesheets/components/_header.scss [hunk 13]
```diff
@@ -0,0 +1,17 @@
...
+  &--multipleCta {
+    @include display(flex);
+
+    a {
+      font-size: 1.4rem;
+      text-transform: uppercase;
+      color: $white;
+    }
+  }
+}
```
Comment (chrislopresto) [inline app/assets/stylesheets/components/_header.scss]: Summing up all of these `sc-TakeoverHeader` comments... I'm not sure we need to touch this component at all. Can we try extracting the link rules and deleting everything else?

**Comment by chrislopresto** [comment]
<< domain tafn

**Comment by agirlnamedsophia** [comment]
bump @chrislopresto

File: app/helpers/application_layout_helper.rb [hunk 14]
```diff
@@ -0,0 +1,56 @@
+module ApplicationLayoutHelper
+  def page_title
+    content_for(:page_title) || 'Test Track Admin'
+  end
+
+  def site_layout_header_classes
```
Comment (chrislopresto) [inline app/helpers/application_layout_helper.rb]: We probably don't need most or all of these site layout helper classes.

**Comment by chrislopresto** [comment]
domain lgtm

**Pull request merged**